### DISCUSSION TITLE
Deploy on Railway

### DISCUSSION THREAD

**Discussion by aileksandar** [directus/directus#9626] [Feature Requests]
I'm currently using [Railway](https://railway.app/) for my Directus instance, and I'm super happy so far. It's based on Heroku buildpacks and IMO its the only real alternative to Heroku on the market.

I think we should consider putting Directus among their [starters](https://railway.app/starters), and also maybe add to readme and docs the "Deploy button" like you have for other providers. I think this can create a big impact for several reasons:
 - It has really good free plan - you can spin development and production instance and you still won't go over the free credit
 - It does not shut down on idle like Heroku
 - Not sure how platform and zeet are used for Directus but they are pretty expensive and thats not a good thing if someone just want to spin and play around with it.
  - The only plus for Digital Ocean that can justify their $5 droplet is that it offers you some SSD for assets while Railway has no storage, on each deploy its gone.
  
Not affiliated, just like the product. Strapi, Ghost and Keystone are already there, I think we should jump in too, I believe this hosting will have a bright future.

**Comment by rijkvanzanten** [comment]
Totally down to have it! @meoweloper Any insights on how to setup a one-click for Railway?

**Reply by aileksandar** [reply]
Glad to hear that 😄  Yeah, they have it in their [docs](https://docs.railway.app/reference/starters). Its fairly simple, they have monorepo with all starters, and you just open a PR with Directus, and they also have a template for the button too.

**Reply by azrikahar** [reply]
Was trying to set it up, and after going in first time, I have to say the platform is pretty solid from my experience.

Tentatively having it at: https://github.com/azrikahar/directus-railway-starter, tested & works.

Here's how it looks like when we click on the button:

(repo name will automatically be `directus-railway-starter` in this case, but since it already exists for me, I changed it to `my-directus-project`)

![chrome_aDX6tj0tOq](https://user-images.githubusercontent.com/42867097/141084859-f23a1190-c114-472b-9824-f269944daccb.png)

@rijkvanzanten there's a distinction depending on where the actual repo is hosted, either on their starter repo directly or link from the starter repo (basically just a readme file) to an actual repo in `directus-community`. Here's a clearer distinction I mentioned:

Notice how there's a community label and border for Calendso & Code Server:

![image](https://user-images.githubusercontent.com/42867097/141083939-df6c7d49-2db0-484b-af4d-1bd5ee0ce249.png)

Example of putting the project directly in there: https://github.com/railwayapp/starters/tree/master/examples/nextjs-prisma

Example of linking from starter to our own repo: https://github.com/railwayapp/starters/tree/master/examples/code-server

Pros on putting it in `directus-community` would feel more "centralized" like the heroku template, but it also depends which starter button "look" is more preferable. Let me know which one do you think is better, and I'll make the PR accordingly.

**Reply by rijkvanzanten** [reply]
If it's a single installer that we don't have to actively maintain (eg it needs to install the latest version on creation automatically), then lets put it on their repo 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by davepoon** [comment]
Recently I created a one-click Directus deployment solution for Railway, As I was unable to find any Directus Railway template that fit my use case for docker and extension development, I decided to create my own. 
You can try this if you

- Prefer using Docker to manage Directus
- S3 persistent storage 
- Prefer PostGIS rather than Postgre
- PostGIS is preconfigured with TCP  
- Saving cost from egress fees

https://railway.app/template/XQc69P?referralCode=OYCuBb
